integration-test/src/test/java/org/jboss/pnc/integration/endpoints/ProductMilestoneEndpointTest.java

```diff
@@ -336,4 +342,54 @@ public void testGetDeliverableAnalyzerOperations() throws ClientException {
     }
+
+    @Test
+    public void testGetStatistics() throws ClientException {
+        // given
+        ProductMilestoneClient client = new ProductMilestoneClient(RestClientConfiguration.asAnonymous());
+
+        DeliveredArtifactsStatistics expectedDeliveredArtifactsStats = DeliveredArtifactsStatistics.builder()
+                .thisMilestone(2) // builtArtifact1, builtArtifact9
+                .otherMilestones(1) // builtArtifact10
+                .otherProducts(2) // builtArtifact11, builtArtifact12
+                .noMilestone(1) // builtArtifact5
+                .noBuild(1) // importedArtifact2
+                .build();
+
+        Supplier<Integer> constantZeroSupplier = () -> 0;
+        EnumMap<ArtifactQuality, Integer> expectedArtifactQualities = EnumMapUtils
+                .initEnumMapWithDefaultValue(ArtifactQuality.class, constantZeroSupplier);
+        expectedArtifactQualities.put(ArtifactQuality.NEW, 6);
+        expectedArtifactQualities.put(ArtifactQuality.VERIFIED, 1);
+
+        EnumMap<RepositoryType, Integer> expectedRepositoryTypes = EnumMapUtils
+                .initEnumMapWithDefaultValue(RepositoryType.class, constantZeroSupplier);
+        expectedRepositoryTypes.put(RepositoryType.MAVEN, 7);
+
+        ProductMilestoneStatistics expectedStats = ProductMilestoneStatistics.builder()
+                .artifactsInMilestone(3) // builtArtifact1, builtArtifact2, builtArtifact9
+                .deliveredArtifactsSource(expectedDeliveredArtifactsStats)
+                .artifactQuality(expectedArtifactQualities)
+                .repositoryType(expectedRepositoryTypes)
+                .build();
+
+        // then
+        ProductMilestoneStatistics actualStats = client.getStatistics(milestoneId);
+
+        // assert
+        assertThat(actualStats.getArtifactsInMilestone()).isEqualTo(expectedStats.getArtifactsInMilestone());
+        assertThat(actualStats.getDeliveredArtifactsSource().getThisMilestone())
+                .isEqualTo(expectedDeliveredArtifactsStats.getThisMilestone());
+        assertThat(actualStats.getDeliveredArtifactsSource().getOtherMilestones())
+                .isEqualTo(expectedDeliveredArtifactsStats.getOtherMilestones());
+        assertThat(actualStats.getDeliveredArtifactsSource().getOtherProducts())
+                .isEqualTo(expectedDeliveredArtifactsStats.getOtherProducts());
+        assertThat(actualStats.getDeliveredArtifactsSource().getNoMilestone())
+                .isEqualTo(expectedDeliveredArtifactsStats.getNoMilestone());
+        assertThat(actualStats.getDeliveredArtifactsSource().getNoBuild())
+                .isEqualTo(expectedDeliveredArtifactsStats.getNoBuild());
+        assertThat(actualStats.getArtifactQuality()).isEqualTo(expectedArtifactQualities);
+        assertThat(actualStats.getRepositoryType()).isEqualTo(expectedRepositoryTypes);
+        assertThat(actualStats).isEqualTo(expectedStats); // also test it as a whole
```

Comment:
Not sure whether we should test it also as a whole. We can forget about some attribute now (or in the future) and this can be good back-up test, which is however quite useless for debugging purposes.